client/js/controllers/user_controller.js

```diff
@@ -21,8 +22,11 @@ class UserController {
             return;
         }
 
+        this._successMessages = [];
+        this._errorMessages = [];
+
         topNavigation.setTitle('User ' + userName);
-        User.get(userName).then(user => {
+        User.get(userName).then(async user => {
```

Comment:
`async`?

Reply:
Ah, that's because I added an await to the internal call for getting the tokens for the user if we're on that view. It needs to happen synchronously before the view is created for that page.

Reply:
I'll need to test this with transpiling back to ES5, I'm not sure if that will work. If it doesn't I'll need to work around that.

Reply:
In case you wind up needing to change it, here you can see how I load models into views.

First, the controller asks for the model:

```
User.get(id)
```

This returns a promise which I can then react to:

```
User.get(id).then(
    user => { success! let's construct the view },
    error => { show the error somewhere... })
```

In some places I also await for multiple resources at once:

```
Promise.all([
    TagCategoryList.get(),
    Tag.get(ctx.parameters.name),
]).then(responses => {
    const [tagCategoriesResponse, tag] = responses;
```

Reply:
I didn't know about `async` and now my promise-based code looks super-ugly. Oh well...

Reply:
It looks like async/await doesn't transpile backwards from es6 to es5 (without a babel plugin), and support has only been common since late 2016 / early 2017 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) So this may be a little new to rely on.

Given that I'll be adapting this to use the Promise.all design. It might not be prettier, but it will be consistent with the codebase and won't increase requirements for the users browser.